src/lib.rs

```diff
@@ -170,28 +175,31 @@ mod tls {
                 |error| InternalConnectError::ReadFile { file: path.into(), error });
             let mut reader = &*contents;
 
-            let mut certs = try_map_err!(rustls_pemfile::certs(&mut reader),
+            let certs = try_map_err!(rustls_pemfile::certs(&mut reader),
                 |error| InternalConnectError::ParseCert { file: path.into(), error });
 
             if certs.len() != 1 {
-                return Err(InternalConnectError::InvalidCertCount { file: path.into(), count: certs.len(), });
+                tracing::debug!("Certificate count is not 1 (Actual Count: {})", certs.len());
             }
 
             Ok(CertVerifier {
-                cert: certs.swap_remove(0),
+                certs: certs,
             })
         }
     }
 
     impl rustls::ServerCertVerifier for CertVerifier {
         fn verify_server_cert(&self, _roots: &RootCertStore, presented_certs: &[Certificate], _dns_name: DNSNameRef<'_>, _ocsp_response: &[u8]) -> Result<ServerCertVerified, TLSError> {
-            if presented_certs.len() != 1 {
-                return Err(TLSError::General(format!("server sent {} certificates, expected one", presented_certs.len())));
-            }
-            if presented_certs[0].0 != self.cert {
-                return Err(TLSError::General(format!("server certificates doesn't match ours")));
+            
+            for (c, p) in self.certs.iter().zip(presented_certs.iter()) {
+                if *p.0 != *c {
+                    return Err(TLSError::General(format!("server certificates do not match ours")));
+                } else {
+                    tracing::trace!("Confirmed certificate match");
+                }
```

Comment:
We don't strictly need this message, but I kind of like having it for sanity checking.